### ISSUE TITLE
Cannot install old version of hie

### ISSUE THREAD

**Issue by drsooch** [haskell/haskell-ide-engine#1763]
I was looking to start the [fp-course](https://github.com/bitemyapp/fp-course) . The GHC version for the project is 8.0.2. I have that GHC installed from stack and everything is all set in that regard. I want to use hie with coc.nvim; however it requires hie-8.0.2 or else I receive a mismatch error for the GHC version. 

I've crawled through closed issues to see if I could resolve this issue and was unable to. The issue I'm running into is when invoking  `stack ./install.hs hie-8.0.2` there is no yaml file available for this version, and an error is thrown. I went back through the commit history and tried to use an old `stack-8.0.2.yaml` file but was unable to get that to work.

In my attempts to fix this issue I reinstalled hie and built it for my current GHC version. When I attempt to run `stack ./install.hs ghcs` to see the available install options it fails stating:
```Error when running Shake build system:
  at want, called at src/Development/Shake/Internal/Args.hs:83:69 in shake-0.18.5-44KSA7uQF2VObxzEvLYZx:Development.Shake.Internal.Args
* Depends on: ghcs
  at error, called at src/Development/Shake/Internal/Rules/File.hs:179:58 in shake-0.18.5-44KSA7uQF2VObxzEvLYZx:Development.Shake.Internal.Rules.File
* Raised the exception:
Error, file does not exist and no rule available:
  ghcs
```

I'm not sure if this is a separate issue.

**Comment by Anrock** [comment]
Support for 8.0.2 was dropped long ago - it didn't even support cabal-2.0. Last commit with 8.0.2 support was c1ca784. Try building hie from that commit. 

> When I attempt to run stack ./install.hs ghcs to see the available install options it fails stating

I don't think there ever was a ghcs target in install.hs.

> I went back through the commit history and tried to use an old stack-8.0.2.yaml file but was unable to get that to work.

What errors exactly?

**Comment by fendor** [comment]
> I don't think there ever was a ghcs target in install.hs.

There is no `ghcs` target in `stack ./install.hs`, only if executed from cabal.

**Comment by drsooch** [comment]
> I don't think there ever was a ghcs target in install.hs.'

I just took a look at the source for HieInstall it looks like when `install.hs` is run with cabal, so that would make sense. Just a little confusing when run from stack it has that option. I would assume `stack ./install.hs help` expands to possible targets?

> What errors exactly?

Basically dependencies I don't care to fix, there's also a parsing error. 

I'll close this out, I can live without hie on a play project.

Thanks
